docs/how-to-use-minio-server-side-encryption.md

```diff
@@ -0,0 +1,65 @@
+# How to use Minio's server-side-encryption [![Slack](https://slack.minio.io/slack?type=svg)](https://slack.minio.io)
+
+Minio supports S3 server-side-encryption with customer provided keys (SSE-C).
+A client **must** specify three HTTP headers for SSE-C requests:
+ - The algorithm identifier: `X-Amz-Server-Side-Encryption-Customer-Algorithm`  
+   The only valid value is: `AES256`.
+ - The secret encryption key: `X-Amz-Server-Side-Encryption-Customer-Key`  
+   The secret encryption key **must** be a 256 bit base64 encoded string.
+ - The encryption key MD5 checksum: `X-Amz-Server-Side-Encryption-Customer-Key-MD5`  
+   The encryption key MD5 checksum **must** be the MD5-sum of the encryption key.
+   The encryption key MD5 checksum is the MD5-sum of the raw binary key not of the
+   base64 encoded key.
```

Comment:
Perhaps you can provide a like to AWS S3 compatibility here?

Reply:
Correct, the problem is that AWS is not explicit about that.
>Use this header to provide the base64-encoded 128-bit MD5 digest of the encryption key according to RFC 1321. Amazon S3 uses this header for a message integrity check to ensure the encryption key was transmitted without error. 

A user may can compute the MD5 of the base64 key which would be incorrect and the server would reject the request with the error msg: `Key does not match Key-MD5` which may confuse the user - because he/she computed MD5 correctly...

Reply:
Isn't rfc1321 supposed to clear that up ?